src/cider/nrepl/middleware/util/java.clj

```diff
@@ -73,9 +73,19 @@
        (str/replace "$" ".")
        (str ".html")))
   ([class member argtypes]
-   (str (javadoc-url class) "#" member
-        (when argtypes
-          (str "(" (str/join ",%20" argtypes) ")")))))
+   (let [java-version (Integer/parseInt util/java-api-version)
+         old-format   (< java-version 8)
+         open-paren   (if old-format "(" "-")
```

Comment:
Yes, I think it's better to just have one `if` where each branch is the entire logic for that version.